user/views.py

```diff
@@ -36,3 +38,54 @@ def form_valid(self, form):
             print(password)
             print(email)
             return HttpResponseForbidden()
+
+class ProfileView(ListView):
+    template_name = "user/profile.html"
+    model = Tweet
+
+    def get_queryset(self):
+        associated_id = self.kwargs['pk']
+        user = User.objects.get(id=associated_id)
+        return Tweet.objects.filter(author=user)
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        associated_id = self.kwargs['pk']
+        context["author"] = User.objects.filter(id=associated_id).first()
+        context["requesting_user"] = self.request.user
+        return context
+
+def follow_view(request, pk):
+    try:
+        follower = User.objects.get(username=request.user.username)
```

Comment:
以下のように`id`ではなく`username`を検索条件にしている理由はありますか？
また、request.userが認証済みであることを前提にしたいのであれば[login_required](https://docs.djangoproject.com/en/4.0/topics/auth/default/#the-login-required-decorator)を用いると良いと思います(その場合、request.userをそのまま使用すれば良いため)
```suggestion
        follower = User.objects.get(id=request.user.id)
```

Reply:
特に理由はありませんでした。`login_required`を使用するように変更しました。